Mathlib/Algebra/Order/BigOperators/Group/LocallyFinite.lean

```diff
@@ -3,78 +3,126 @@ Copyright (c) 2024 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
-import Mathlib.Order.Interval.Finset.Basic
-import Mathlib.Algebra.BigOperators.Group.Finset.Basic
+import Mathlib.Algebra.BigOperators.Group.Finset.Sigma
+import Mathlib.Algebra.Order.Interval.Finset.SuccPred
+import Mathlib.Data.Nat.SuccPred
+import Mathlib.Order.Interval.Finset.Nat
 
 /-!
 # Big operators indexed by intervals
 
 This file proves lemmas about `∏ x ∈ Ixx a b, f x` and `∑ x ∈ Ixx a b, f x`.
 -/
 
-variable {α β : Type*} [PartialOrder α] [CommMonoid β] {f : α → β} {a b : α}
+open Order
+
+variable {α M : Type*} [CommMonoid M] {f : α → M} {a b : α}
 
 namespace Finset
+section PartialOrder
+variable [PartialOrder α]
+
 section LocallyFiniteOrder
 variable [LocallyFiniteOrder α]
 
 @[to_additive]
-lemma left_mul_prod_Ioc (h : a ≤ b) : f a * ∏ x ∈ Ioc a b, f x = ∏ x ∈ Icc a b, f x := by
-  rw [Icc_eq_cons_Ioc h, prod_cons]
+lemma mul_prod_Ico_eq_prod_Icc (h : a ≤ b) : f b * ∏ x ∈ Ico a b, f x = ∏ x ∈ Icc a b, f x := by
+  rw [Icc_eq_cons_Ico h, prod_cons]
 
-@[to_additive]
-lemma prod_Ioc_mul_left (h : a ≤ b) : (∏ x ∈ Ioc a b, f x) * f a = ∏ x ∈ Icc a b, f x := by
-  rw [mul_comm, left_mul_prod_Ioc h]
+@[deprecated (since := "2025-05-03")] alias right_mul_prod_Ico := mul_prod_Ico_eq_prod_Icc
```

Comment:
Don't forget to deprecate the additive versions too.

Reply:
It's honestly a bit too much work for an API that was introduced right before that PR